ContactDetailsApi/V2/Controllers/ServicesoftController.cs

```diff
@@ -20,11 +21,16 @@ public ServicesoftController(IFetchAllContactDetailsByUprnUseCase fetchAllContac
             _fetchAllContactDetailsByUprnUseCase = fetchAllContactDetailsByUprnUseCase;
         }
 
+        /// <summary>
+        /// Retrieves contact details for a slice of tenures grouped by UPRN
+        /// </summary>
         [HttpGet]
         [Route("contactDetails")]
         [AuthorizeEndpointByIpWhitelist("WHITELIST_IP_ADDRESS")]
         public async Task<IActionResult> FetchAllContactDetailsByUprn([FromQuery] ServicesoftFetchContactDetailsRequest request)
         {
+            // Limit the page size to avoid API Gateway timeout (30s)
+            request.PageSize = Math.Clamp(request.PageSize, 1, 500);
```

Comment:
Can't you use a validator for this instead of enforcing it in the controller? Something like this should work:

```csharp
using FluentValidation;

namespace ContactDetailsApi.V2.Boundary.Request.Validation
{
    public class ServicesoftFetchContactDetailsRequestValidator : AbstractValidator<ServicesoftFetchContactDetailsRequest>
    {
        public ServicesoftFetchContactDetailsRequestValidator()
        {
            RuleFor(x => x.PageSize).InclusiveBetween(1, 500)
                                    .When(y => y.PageSize.HasValue);
        }
    }
}
```

Reply:
Won't that throw an error if it's outside the range though?

Reply:
The PageSize won't correspond with the amount of results they actually get so it's somewhat misleading in the first place

Reply:
Maybe the idea has changed since we caught up yesterday, but I thought we were only adding PageSize so that we could use it for testing purposes.

Servicesoft want to get all the data as fast as possible, so they would likely not use the PageSize field at all, which would then default to the maximum page size allowed (whatever we define). This would just mean that if they choose to specify the pagesize in the request, it has to between our defined range.

Reply:
> "The PageSize won't correspond with the amount of results they actually get so it's somewhat misleading in the first place"

@adamtry What do you mean?
So if I say, I want 400 results per page (PageSize=400), how many will I get back?